pkg/kubectl/cmd/config/list_context.go

```diff
@@ -0,0 +1,58 @@
+/*
+Copyright 2014 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package config
+
+import (
+	"fmt"
+	"io"
+
+	"github.com/spf13/cobra"
+	"k8s.io/kubernetes/pkg/client/unversioned/clientcmd"
+
+	cmdutil "k8s.io/kubernetes/pkg/kubectl/cmd/util"
+)
+
+type ListContextOptions struct {
+	configAccess clientcmd.ConfigAccess
+}
+
+func NewCmdConfigListContext(out io.Writer, configAccess clientcmd.ConfigAccess) *cobra.Command {
+	options := &ListContextOptions{configAccess: configAccess}
+
+	cmd := &cobra.Command{
+		Use:   "list-context",
```

Comment:
Please call it `get-context`. `list` is not an ordinary kubectl command, but `get` is.

`kubectl get-context` should print all contexts.

`kubectl get-context <context-name>` should print just the selected context.

`kubectl get-context -o name` should print the names of all contexts.